src/gmail.js

```diff
@@ -3442,17 +3442,28 @@ var Gmail = function(localJQuery) {
         }
 
         if (typeof element === "string") {
-            this.id = element;
-            element = $("div.adn[data-legacy-message-id='" + this.id + "']");
+            if (element.substring(0,3) === 'msg') {
+                this.id = element;
+                element = $("div.adn[data-message-id='#" + this.id + "']");
+            } else {
+                this.legacy_email_id = element;
+                element = $("div.adn[data-legacy-message-id='" + this.legacy_email_id + "']");
+            }   
         } else {
             element = $(element);
         }
 
         if (!element || (!element.hasClass("adn"))) api.tools.error("api.dom.email called with invalid element/id");
 
-        this.$el = element;
-        if (!this.id) {
-            this.id = this.$el.data("legacyMessageId");
+        this.$el = element;        
+        if (!this.legacy_email_id) {
+            this.legacy_email_id = this.$el.data("legacyMessageId");
+        }
+        if (!this.id) {            
+            let email_id = this.$el.data("messageId");
+            if (email_id) {
+                this.id = email_id.substring(1);   
+            }         
```

Comment:
From what I can tell, this change breaks compatibility by changing what the `.id`-property represents.

I'll be happy to add new-style ID properties, but only in addition to the existing `.id`-property which for compatibility-reasons I'd like to remain as it is.

Is that OK with you?